src/typechecker.erl

```diff
@@ -3463,6 +3463,8 @@ refinable(?type(union,Tys)) when is_list(Tys) ->
     lists:all(fun refinable/1, Tys);
 refinable(?type(tuple, Tys)) when is_list(Tys) ->
     lists:all(fun refinable/1, Tys);
+refinable(?type(record, [_ | Fields])) ->
+    lists:all(fun refinable/1, [X || ?type(field_type, X) <- Fields]);
```

Comment:
Do the Fields here contain all the fields of the record type or only the refined ones? (@FrankBro do you know?) If now, we would need to pass Env here to be able to get the non-refined ones and check if they are refinable as well.

Reply:
From the testing I have done, The list of fields has always been empty.

Reply:
Ah. Then, only the refined fields are included in Fields here.

Given a record definition

```Erlang
-record(rec, {foo :: a | b | c, bar :: float()}).
```

... the type `#rec{}` means the full record (Fields empty) and `#rec{foo :: c}` is a refined record type (Fields contains foo :: c). `bar` is not included because it's the same as in the record definition and is not printed when you print the record type. But because `float()` is not refinable, neither `#rec{}` nor `#rec{foo :: c}` are refinable.

Perhaps it would be good to include a test case with a non-refinable field in a record. If we're not handling it correcty, just place it under known_problems.